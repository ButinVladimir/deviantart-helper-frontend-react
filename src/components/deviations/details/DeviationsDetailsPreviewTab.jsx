import React from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Level from 'react-bulma-components/lib/components/level';

/* eslint-disable react/no-danger */

export default function DeviationsDetailsPreviewTab({
  preview,
  title,
}) {
  return (
    <Section>
      <Container>
        <Level>
          <Level.Item>
            <img alt={title} src={preview.src} />
          </Level.Item>
        </Level>
      </Container>
    </Section>
  );
}

DeviationsDetailsPreviewTab.propTypes = {
  preview: PropTypes.shape({
    src: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired,
  title: PropTypes.string.isRequired,
};

/* eslint-enable no-danger */
