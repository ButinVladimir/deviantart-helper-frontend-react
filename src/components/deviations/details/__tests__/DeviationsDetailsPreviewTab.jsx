import React from 'react';
import { shallow } from 'enzyme';
import DeviationsDetailsPreviewTab from '../DeviationsDetailsPreviewTab';

describe('DeviationsDetailsPreviewTab', () => {
  it('can be rendered correctly', () => {
    const wrapper = shallow(
      <DeviationsDetailsPreviewTab
        preview={{
          src: '/path/to/image',
          width: 3,
          height: 4,
        }}
        title="Deviation title"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
