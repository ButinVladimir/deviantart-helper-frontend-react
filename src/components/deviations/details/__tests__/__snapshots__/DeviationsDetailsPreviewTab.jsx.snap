// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsDetailsPreviewTab can be rendered correctly 1`] = `
<Section
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <Container
    breakpoint={null}
    fluid={false}
    renderAs="div"
    style={Object {}}
  >
    <Level
      breakpoint={null}
      className=""
      renderAs="div"
      style={Object {}}
    >
      <LevelItem
        className=""
        renderAs="div"
        style={Object {}}
      >
        <img
          alt="Deviation title"
          src="/path/to/image"
        />
      </LevelItem>
    </Level>
  </Container>
</Section>
`;
